TDDWithURMAndKotlin/part2_increment.md

```diff
@@ -0,0 +1,157 @@
+# TDD With URM and Kotlin - increment
+
+In the previous article we mentioned a way to start writing our implementation of URM using TDD. We also mentioned a lot of questions that came up while writing the `zero` function.
+
+Now it's time to write the increment function.
+
+## Increment function
+
+We know the `increment` function will be `I(x)`, where `x` represents a register in our _Unlimented Register Machine_. We are writing (for now) only the internal that will be used to represent our functionallity.
+
+Following the approach we followed for the `zero` function, we start by writing a test on how we expect the function behaves with negative numbers:
+
+```kotlin
+@ParameterizedTest
+@ValueSource(ints = [-2, -8, -20, -100])
+fun `increment function throws exception with negative position`(position: Int) {
+    assertThrows<IllegalArgumentException> { increment(registry, position) }
+
+    assertNull(registry.getValueAtPosition(position))
+}
+```
+
+This initial test is in pro of the defensive model, we first need to be sure the function fails for some invalid cases. We could start with a different part of the code but this is the easiest to write because we already wrote this on the `zero` function:
+
+```kotlin
+fun increment(registry: Registry, position: Int){
+    require(position > 0) { "Position must be positive number" }
+}
+```
+
+For our second test we need want to validate a "strict" rule: when a register has no value, it cannot be incremented. Our test should represent this and we should put a simple exception for it:
```

Comment:
- For our second test, **(coma)** we need...

- this is not quiet understandable "we need want to validate". take off need or want

Reply:
I removed the `want` because we actually `need`